src/db_adapters/access_keys.rs

```diff
@@ -12,108 +10,58 @@ use crate::schema;
 
 pub(crate) async fn handle_access_keys(
     pool: &actix_diesel::Database<PgConnection>,
-    outcomes: &[near_lake_framework::near_indexer_primitives::IndexerExecutionOutcomeWithReceipt],
+    state_changes: &[near_lake_framework::near_indexer_primitives::views::StateChangeWithCauseView],
     block_height: near_lake_framework::near_indexer_primitives::types::BlockHeight,
 ) -> anyhow::Result<()> {
-    if outcomes.is_empty() {
+    if state_changes.is_empty() {
         return Ok(());
     }
-    let successful_receipts = outcomes
-        .iter()
-        .filter(|outcome_with_receipt| {
-            matches!(
-                outcome_with_receipt.execution_outcome.outcome.status,
-                near_lake_framework::near_indexer_primitives::views::ExecutionStatusView::SuccessValue(_)
-                    | near_lake_framework::near_indexer_primitives::views::ExecutionStatusView::SuccessReceiptId(_)
-            )
-        })
-        .map(|outcome_with_receipt| &outcome_with_receipt.receipt);
 
     let mut access_keys = HashMap::<(String, String), models::access_keys::AccessKey>::new();
-    let mut deleted_accounts = HashMap::<String, String>::new();
 
-    for receipt in successful_receipts {
-        if let near_lake_framework::near_indexer_primitives::views::ReceiptEnumView::Action {
-            actions,
-            ..
-        } = &receipt.receipt
+    for state_change in state_changes {
+        if let near_lake_framework::near_indexer_primitives::views::StateChangeCauseView::ReceiptProcessing { receipt_hash } =
+            state_change.cause
         {
-            for action in actions {
-                match action {
-                    near_lake_framework::near_indexer_primitives::views::ActionView::DeleteAccount { .. } => {
```

Comment:
What's the reasoning for using `near_lake_framework::near_indexer_primitives::views` over `near_primitives::views::`?

Reply:
This is to avoid potential version mismatch when your Cargo.toml uses one version while `near-lake-framework` stick to a different semver-incompatible version.